frontend/src/container/TracesExplorer/ListView/index.tsx

```diff
@@ -124,16 +131,36 @@ function ListView(): JSX.Element {
 
 	return (
 		<Container>
-			<TraceExplorerControls
-				isLoading={isFetching}
-				totalCount={totalCount}
-				config={config}
-				perPageOptions={PER_PAGE_OPTIONS}
-			/>
+			{transformedQueryTableData.length !== 0 && (
+				<TraceExplorerControls
+					isLoading={isFetching}
+					totalCount={totalCount}
+					config={config}
+					perPageOptions={PER_PAGE_OPTIONS}
+				/>
+			)}
 
 			{isError && <ErrorText>{data?.error || 'Something went wrong'}</ErrorText>}
 
-			{!isError && (
+			{(isLoading || (isFetching && transformedQueryTableData.length === 0)) && (
+				<TracesLoading />
+			)}
+
+			{!isLoading &&
```

Comment:
We should move this to a single variable and list them next to each other for better readability of states

Reply:
Done